arangod/Indexes/Index.cpp

```diff
@@ -413,18 +413,28 @@ bool Index::validateHandle(char const* key, size_t* split) {
 /// @brief generate a new index id
 TRI_idx_iid_t Index::generateId() { return TRI_NewTickServer(); }
 
+/// @brief check if two index definitions share any identifiers (_id, name)
+bool Index::CompareIdentifiers(velocypack::Slice const& lhs, velocypack::Slice const& rhs) {
+  VPackSlice lhsId = lhs.get(arangodb::StaticStrings::IndexId);
+  VPackSlice rhsId = rhs.get(arangodb::StaticStrings::IndexId);
+  if (lhsId.isString() && rhsId.isString() &&
+      arangodb::basics::VelocyPackHelper::compare(lhsId, rhsId, true) == 0) {
+    return true;
+  }
+
+  VPackSlice lhsName = lhs.get(arangodb::StaticStrings::IndexName);
+  VPackSlice rhsName = rhs.get(arangodb::StaticStrings::IndexName);
+  if (lhsName.isString() && rhsName.isString() &&
+      arangodb::basics::VelocyPackHelper::compare(lhsName, rhsName, true) == 0) {
+    return true;
+  }
+
+  return false;
+}
+
 /// @brief index comparator, used by the coordinator to detect if two index
 /// contents are the same
 bool Index::Compare(VPackSlice const& lhs, VPackSlice const& rhs) {
-  {
```

Comment:
This is a public static method and is also called from here:
```
arangod/Cluster/Maintenance.cpp:              if (!arangodb::Index::Compare(pindex, lindex)) {
arangod/ClusterEngine/ClusterIndex.cpp:  return Index::Compare(_info.slice(), info);
```
Do the above place need adjustment too?

Reply:
Nope, these are okay to use the old check.